### PR TITLE
Fix http status codes

### PR DESCRIPTION
**Pull request by i1i1**
Signed-off-by: i1i1 <vanyarybin1@live.ru>

### Description of the Change

Http status code now are working.

### Issue

Before hand every error status code was 500, because of weird warp crate architecture.

### Benefits

Proper error reporting in http

### CODE REVIEW

File: iroha/src/torii/mod.rs [hunk 1]
```diff
@@ -138,6 +139,17 @@ impl<W: WorldTrait> Torii<W> {
...
+        #[allow(clippy::unused_async)]
+        async fn recover(err: Rejection) -> Result<impl Reply, Rejection> {
+            if let Some(err) = err.find::<AcceptQueryError>() {
+                return Ok(reply::with_status(err.to_string(), err.status_code()));
+            }
+            if let Some(err) = err.find::<iroha_version::error::Error>() {
+                return Ok(reply::with_status(err.to_string(), err.status_code()));
+            }
+            Err(err)
+        }
```
Comment (eadventurous) [inline iroha/src/torii/mod.rs:151]: Maybe I missed something, can you tell me where unused async is?

File: iroha/src/torii/mod.rs [hunk 2]
```diff
@@ -138,6 +139,17 @@ impl<W: WorldTrait> Torii<W> {
...
     #[iroha_futures::telemetry_future]
     pub async fn start(self) -> iroha_error::Result<()> {
+        #[allow(clippy::unused_async)]
+        async fn recover(err: Rejection) -> Result<impl Reply, Rejection> {
+            if let Some(err) = err.find::<AcceptQueryError>() {
+                return Ok(reply::with_status(err.to_string(), err.status_code()));
+            }
+            if let Some(err) = err.find::<iroha_version::error::Error>() {
+                return Ok(reply::with_status(err.to_string(), err.status_code()));
+            }
```
Comment (eadventurous) [inline iroha/src/torii/mod.rs:149]: Do I understand correctly that this are errors for which we return custom Error http codes?

File: iroha/src/torii/mod.rs [hunk 3]
```diff
@@ -138,6 +139,17 @@ impl<W: WorldTrait> Torii<W> {
     /// Can fail due to listening to network or if http server fails
     #[iroha_futures::telemetry_future]
     pub async fn start(self) -> iroha_error::Result<()> {
+        #[allow(clippy::unused_async)]
+        async fn recover(err: Rejection) -> Result<impl Reply, Rejection> {
```
Comment (eadventurous) [inline iroha/src/torii/mod.rs:143]:
```suggestion
        async fn custom_http_error_code(err: Rejection) -> Result<impl Reply, Rejection> {
```

Comment (eadventurous) [reply iroha/src/torii/mod.rs:143]: Or something like that. Maybe also leave a comment on top of the function there.

File: iroha/src/torii/mod.rs (hunk 1 above)
Comment (i1i1) [reply iroha/src/torii/mod.rs:151]: We dont do `.await` here

File: iroha/src/torii/mod.rs (hunk 2 above)
Comment (i1i1) [reply iroha/src/torii/mod.rs:149]: Thats all errors in parsing arguments

File: iroha/src/torii/mod.rs (hunk 1 above)
Comment (eadventurous) [reply iroha/src/torii/mod.rs]: `with_status` returns a future?

**Review by eadventurous — DISMISSED** [review]

File: iroha/src/torii/mod.rs (hunk 1 above)
Comment (i1i1) [reply iroha/src/torii/mod.rs]: Nope. This lint says that function could be sync

**Review by eadventurous — APPROVED** [review]

**Pull request merged**